test/rounding.jl

```diff
@@ -351,3 +351,10 @@ end
         Base.Rounding.setrounding_raw(T, Base.Rounding.to_fenv(old))
     end
 end
+
+@testset "rounding to AbstractFloat" begin
+    for T in [Float16, Float32, Float64]
+        @test floor(T, 2.3) <= 2.3
```

Comment:
Testing this is probably not enough, we also need to test that this doesn't return `floor(2.3) == 2.0`.
It's also worth adding a comment about the fact that `Float16(2.3) > 2.3` and `Float32(2.3) < 2.3`.